src/cargo/core/registry.rs

```diff
@@ -779,7 +804,20 @@ fn summary_for_patch(
     // No summaries found, try to help the user figure out what is wrong.
     if let Some((_locked_patch, locked_id)) = locked {
         // Since the locked patch did not match anything, try the unlocked one.
-        let orig_matches = source.query_vec(orig_patch).unwrap_or_else(|e| {
+        let orig_matches = loop {
+            match source.query_vec(orig_patch) {
+                Ok(Poll::Ready(deps)) => {
+                    break Ok(deps);
+                }
+                Ok(Poll::Pending) => {
+                    // TODO: dont hot loop for it to be Ready
```

Comment:
I think that this method may want to be refactored to return a `Poll` up to the `patch` method which would handle the blocking-on-sources.

Reply:
Good call. Moved it up.